packages/contract/src/contracts/functions/invocation-results.ts

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable max-classes-per-file */
+import type {
+  CallResult,
+  ReceiptScriptResult,
+  TransactionResponse,
+  TransactionResult,
+} from '@fuel-ts/providers';
+import { ReceiptType } from '@fuel-ts/providers';
+
+import { contractCallScript } from '../../scripts';
+import type { InvocationScopeLike } from '../../types';
+
+function getGasUsage(callResult: CallResult) {
+  const scriptResult = callResult.receipts.find((r) => r.type === ReceiptType.ScriptResult) as
+    | ReceiptScriptResult
+    | undefined;
+  return scriptResult?.gasUsed || 0n;
+}
+
+class InvocationResult<T = any> {
+  readonly functionScopes: Array<InvocationScopeLike>;
+  readonly isMultiCall: boolean;
+  readonly gasUsed: bigint;
+  readonly value: T;
+
+  constructor(
+    funcScopes: InvocationScopeLike | Array<InvocationScopeLike>,
+    callResult: CallResult,
+    isMultiCall: boolean
+  ) {
+    this.functionScopes = Array.isArray(funcScopes) ? funcScopes : [funcScopes];
+    this.isMultiCall = isMultiCall;
+    this.value = this.getDecodedValue(callResult);
+    this.gasUsed = getGasUsage(callResult);
+  }
+
+  protected getDecodedValue(callResult: CallResult) {
+    const encodedResults = contractCallScript.decodeCallResult(callResult);
+    const returnValues = encodedResults.map((encodedResult, i) => {
+      const { contract, func } = this.functionScopes[i].getCallConfig();
+      return contract.interface.decodeFunctionResult(func, encodedResult)?.[0];
+    });
+    return (this.isMultiCall ? returnValues : returnValues?.[0]) as T;
```

Comment:
No, because if you mount a multi-call with a single call method, which can be useful in cases where you dynamically mount the number of calls. In this case, the SDK user would need to also check if the value is an array or not.

Knowing that is a multi-call or single makes it easier to have consistency. Also for the user is a transparent process. as the multi-call and single-call upper classes set this.